tutorials/preprocessing/plot_20_rejecting_bad_data.py

```diff
@@ -1,202 +1,223 @@
-# parameter is named *reject*. It is a dictionary where every key is a
-# channel type as a string and the corresponding values are peak-to-peak
-# rejection parameters (amplitude ranges as floats). Below we define
-# the peak-to-peak rejection values for gradiometers,
-# magnetometers and EOG:
+# Notice that we've passed ``reject_by_annotation=False`` above, in order to
+# isolate the effects of the rejection thresholds. If we re-run the epoching
+# with ``reject_by_annotation=True`` (the default) we see that the rejections
+# due to EEG and EOG channels have disappeared (suggesting that those channel
+# fluctuations were probably blink-related, and were subsumed by rejections
+# based on the "bad blink" label).
 
-reject = dict(grad=4000e-13, mag=4e-12, eog=150e-6)
+epochs = mne.Epochs(raw, events, tmin=-0.2, tmax=0.5, reject_tmax=0,
+                    reject=reject_criteria, flat=flat_criteria, preload=True)
+epochs.plot_drop_log(color='0.5')
 
 ###############################################################################
-# .. note::
-#    The rejection values can be highly data dependent. You should be careful
-#    when adjusting these values. Make sure not too many epochs are rejected
-#    and look into the cause of the rejections. Maybe it's just a matter
-#    of marking a single channel as bad and you'll be able to save a lot
-#    of data.
+# More importantly, note that *many* more epochs are rejected (~20% instead of
+# ~2.5%) when rejecting based on the blink labels, underscoring why it is
+# usually desirable to repair artifacts rather than exclude them.
+#
+# The :meth:`~mne.Epochs.plot_drop_log` method is a visualization of an
+# :class:`~mne.Epochs` attribute, namely ``epochs.drop_log``, which stores
+# empty lists for retained epochs and lists of strings for dropped epochs, with
+# the strings indicating the reason(s) why the epoch was dropped. For example:
 
-###############################################################################
-# We then construct the epochs
-events = mne.find_events(raw, stim_channel='STI 014')
-event_id = {"auditory/left": 1}
-tmin = -0.2  # start of each epoch (200ms before the trigger)
-tmax = 0.5  # end of each epoch (500ms after the trigger)
-baseline = (None, 0)  # means from the first instant to t = 0
-epochs = mne.Epochs(raw, events, event_id, tmin, tmax, proj=True,
-                    picks=['meg', 'eog'], baseline=baseline, reject=reject,
-                    reject_by_annotation=True)
+print(epochs.drop_log)
 
 ###############################################################################
-# We then drop/reject the bad epochs
+# Finally, it should be noted that just like channels marked as bad, "dropped"
+# epochs are not actually deleted from the :class:`~mne.Epochs` object right
+# away. Instead, they are only deleted when the user calls the
+# :meth:`~mne.Epochs.drop_bad` method. If ``reject`` and/or ``flat`` criteria
```

Comment:
hmm ... the story is a bit more complicated than that. If `preload=True` then the epochs are dropped right away, no? If on the other hand, `preload=False`, the epochs are dropped when you try to access them e.g., with `epochs.load_data()`. Finally, you can also give more stringent parameters than what is used in the constructor using `epochs.drop_bad`.